src/build.c

```diff
@@ -826,6 +826,23 @@ static gchar *prepare_run_cmd(GeanyDocument *doc, gchar **working_dir, guint cmd
 	}
 #endif
 
+#ifdef G_OS_WIN32
+	/* Expand environment variables like %blah%. */
+	SETPTR(cmd_string, win32_expand_environment_variables(cmd_string));
+
+	gchar *helper = g_build_filename(utils_resource_dir(RESOURCE_DIR_LIBEXEC), "geany-run-helper", NULL);
+	gchar *arg_directory = NULL;
+
+	/* escape helper appropriately */
+	/* FIXME: check the Windows rules, but it should not matter too much here as \es and "es are not
+	 * allowed in paths anyway */
+	SETPTR(helper, g_strdup_printf("\"%s\"", helper));
+	SETPTR(arg_directory, g_strdup_printf("\"%s\"", *working_dir));
+
+	run_cmd = g_strdup_printf("%s %s %d %s", helper, arg_directory, autoclose ? 1 : 0, cmd_string);
```

Comment:
here you could use the simpler `"\"%s\" \"%s\" %d %s", helper, *working_dir, ...` for the moment